### PR TITLE
Moved mine layer bot module to commons and polish

### PR DESCRIPTION
**Pull request by Mailaender**
Followup to https://github.com/OpenRA/OpenRA/pull/20574 and https://github.com/OpenRA/OpenRA/pull/20933 including some changes from @OpenHV that I applied to make it more readable to me.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs [hunk 1]
```diff
@@ -87,7 +85,7 @@ public MinelayerBotModule(Actor self, MinelayerBotModuleInfo info)
 			world = self.World;
 			player = self.Owner;
 			unitCannotBeOrdered = a => a == null || a.IsDead || !a.IsInWorld || a.Owner != player;
-			unitCannotBeOrderedOrIsBusy = a => unitCannotBeOrdered(a) || (!a.IsIdle && !(a.CurrentActivity is FlyIdle));
+			unitCannotBeOrderedOrIsBusy = a => unitCannotBeOrdered(a) || !a.IsIdle;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs:88]: Why was this removed?

File: OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs [hunk 2]
```diff
@@ -31,16 +28,17 @@ public class MinelayerBotModuleInfo : ConditionalTraitInfo
 		[Desc("Victim target types that considering conflict location as enemy location instead of victim location.")]
 		public readonly BitSet<TargetableType> UseEnemyLocationTargetTypes = default;
 
-		[Desc("Actor types that used for mine laying, must have Minelayer.")]
-		public readonly HashSet<string> Minelayers = default;
+		[ActorReference(typeof(MinelayerInfo))]
+		[Desc("Actors with " + nameof(Minelayer) + "trait.")]
+		public readonly HashSet<string> MinelayingActorTypes = default;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs:33]: This will need an update rule.

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/07c8a0e78dc112aea30563f85d4b1b1ca7ef1662...0edf0a2cc8a4d0b2eca8334553ef27702c6d4a09)

**Comment by PunkPun** [comment]
This is a new trait so it doesn't need update rules

File: OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs (hunk 1 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs:88]: I'm not sure why it was added here in the first place

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs:88]: I figured that a mine laying aircraft would be very odd and removed it for performance reasons.

File: OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs (hunk 2 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/MinelayerBotModule.cs:33]: We never released with https://github.com/OpenRA/OpenRA/pull/20574.
